src/frameworks/nuxt2/index.ts

```diff
@@ -0,0 +1,130 @@
+import { copy, pathExists } from "fs-extra";
+import { readFile } from "fs/promises";
+import { join } from "path";
+import { lt } from "semver";
+import { findDependency, FrameworkType, relativeRequire, SupportLevel } from "..";
+
+export const name = "Nuxt 2";
+export const support = SupportLevel.Experimental;
+export const type = FrameworkType.Toolchain;
+
+export async function discover(dir: string) {
+  if (!(await pathExists(join(dir, "package.json")))) return;
+  const nuxtDependency = findDependency("nuxt", { cwd: dir, depth: 0, omitDev: false });
+  const version = nuxtDependency?.version;
+  const configFilesExist = await Promise.all([
+    pathExists(join(dir, "nuxt.config.js")),
+    pathExists(join(dir, "nuxt.config.ts")),
+  ]);
+  const anyConfigFileExists = configFilesExist.some((it) => it);
+
+  if (!anyConfigFileExists && !nuxtDependency) return;
+  if (!version) throw new Error("Unable to find the nuxt dep.");
+  if (lt(version, "3.0.0")) return { mayWantBackend: true };
+
+  return;
+}
+
+export async function build(root: string) {
+  const nuxt = await getNuxtAppForBuild(root);
+
+  const nuxtApp = await nuxt.loadNuxt({
+    for: "build",
+    rootDir: root,
+  });
+
+  const {
+    options: { target },
+  } = await nuxt.build(nuxtApp);
+
+  if (target === "static") {
+    const nuxtApp = await nuxt.loadNuxt({
+      for: "start",
+      rootDir: root,
+    });
+
+    const builder = await nuxt.getBuilder(nuxtApp);
+    const generator = new nuxt.Generator(nuxtApp, builder);
+    await generator.generate({ build: false, init: true });
+  } else {
+    return { wantsBackend: true };
+  }
+
+  return;
+}
+
+/**
+ * Get the Nuxt app
+ * @param cwd
+ * @return Nuxt app object
+ */
+async function getNuxtAppForBuild(cwd: string) {
+  let nuxt: any = null;
+  try {
```

Comment:
IMO we should be able to drop the try/catch

Reply:
implemented a small refactor; removed the try/catch block